spec/darwin/cpu.go

```diff
@@ -21,6 +25,54 @@ var cpuLogger = logging.GetLogger("spec.cpu")
 
 type cpuSpec map[string]interface{}
 
+var sysCtlKeyMap = map[string]string{
+	"core_count":   "cores",
+	"brand_string": "model_name",
+	"model":        "model",
+	"vendor":       "vendor_id",
+	"family":       "family",
+	"stepping":     "stepping",
+}
+
+func (g *CPUGenerator) parseSysCtlBytes(res []byte) (interface{}, error) {
+	scanner := bufio.NewScanner(bytes.NewBuffer(res))
+
+	results := cpuSpec{}
+	var cores int64
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		kv := strings.SplitN(line, ":", 2)
+		if len(kv) < 2 {
+			continue
+		}
+		key := strings.TrimPrefix(strings.TrimSpace(kv[0]), "machdep.cpu.")
+		val := strings.TrimSpace(kv[1])
+		if label, ok := sysCtlKeyMap[key]; ok {
+			results[label] = val
+		}
+		if key == "core_count" {
+			var err error
+			cores, err = strconv.ParseInt(val, 10, 32)
+			if err != nil {
+				cpuLogger.Errorf("while parsing %q: %s", val, err)
+				return nil, err
+			}
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		cpuLogger.Errorf("Failed (skip this spec): %s", err)
+		return nil, err
+	}
+
+	wholeResults := []cpuSpec{}
```

Comment:
Allocating necessary size of slice is preferable (`make([]cpuSpec{}, 0, coreCount)`) when its capacity is known before appending.